google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ConnectionImpl.java

```diff
@@ -190,11 +190,13 @@ public BigQueryResult executeSelect(String sql) throws BigQuerySQLException {
     try {
```

Comment:
May be better for another PR, but why is all this logic duplicated here, rather than just using a common executeSelect and not supplying the optional params/labels/etc?

Reply:
Thanks for reviewing Seth, I have captured refactoring this as a task @ https://github.com/googleapis/java-bigquery/issues/2240 point#3 . Will update you once we raise a PR.